subprojects/shipkit/src/test/groovy/org/shipkit/internal/gradle/util/CannotPushToGithubExceptionTest.groovy

```diff
@@ -0,0 +1,47 @@
+package org.shipkit.internal.gradle.util
+
+import org.gradle.api.GradleException
+import org.shipkit.gradle.git.GitPushTask
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class CannotPushToGithubExceptionTest extends Specification {
+
+    def "should throw proper message for lack of GH_WRITE_TOKEN"() {
+        given:
+        GitPushTask gitPushTask = Mock(GitPushTask)
+        Exception originalException = new RuntimeException("Exception message")
+        when:
+        def result = CannotPushToGithubException.create(originalException, gitPushTask)
+        then:
+        result.message == CannotPushToGithubException.GH_WRITE_TOKEN_NOT_SET_MSG
```

Comment:
Let's test if we are correctly propagating exception 'cause'. This way we have full coverage for the behavior. Applies to other tests, too.